### DISCUSSION TITLE
Maui app crash in release mode while in debugg its working fine

### DISCUSSION THREAD

**Discussion by UmairHashmi90** [dotnet/maui#23193] [Q&A]
in my android option package formate for debugger apk and release for bundle
fast development is check 
genrate per ABI is uncheck
use incremnetal packaging is check
multidex is uncheck 
 R8 code shrinker is uncheck 
Developer instrumentation is check for debug and uncheck for release 
AOT is check for Release 
LLVM is uncheck
start traching is check for relaese
garbage collection is check 
enable trimming is check for relase
Triming Granularity is link 
java max heap is 1g 

android manafist 
<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android" android:versionCode="23" android:versionName="23">
	<application android:allowBackup="true" android:supportsRtl="true" android:label="KIH Mobile App" android:icon="@mipmap/appicon"></application>
	<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
	<uses-permission android:name="android.permission.INTERNET" />
</manifest>

proejct proerties

<Project Sdk="Microsoft.NET.Sdk">

	<PropertyGroup>
		<TargetFrameworks>net8.0-android;net8.0-ios;net8.0-maccatalyst</TargetFrameworks>
		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);net8.0-windows10.0.19041.0</TargetFrameworks>
		
		

		
		

		<OutputType>Exe</OutputType>
		<RootNamespace>PatientApp</RootNamespace>
		<UseMaui>true</UseMaui>
		<SingleProject>true</SingleProject>
		<ImplicitUsings>enable</ImplicitUsings>
		<Nullable>enable</Nullable>

		
		<ApplicationTitle>KIH Mobile App</ApplicationTitle>

		
		<ApplicationId>com.kihmobile.kih</ApplicationId>

		
		<ApplicationDisplayVersion>1.0</ApplicationDisplayVersion>
		<ApplicationVersion>1</ApplicationVersion>

		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">11.0</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'maccatalyst'">13.1</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">21.0</SupportedOSPlatformVersion>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</SupportedOSPlatformVersion>
		<TargetPlatformMinVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'windows'">10.0.17763.0</TargetPlatformMinVersion>
		<SupportedOSPlatformVersion Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'tizen'">6.5</SupportedOSPlatformVersion>
	
		
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net8.0-android|AnyCPU'">
	  <TrimMode>link</TrimMode>
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net8.0-android|AnyCPU'">
	  <TrimMode>link</TrimMode>
	</PropertyGroup>
	
	<ItemGroup>
		
		<MauiIcon Include="Resources\AppIcon\appicon.svg" ForegroundFile="Resources\AppIcon\appiconfg.svg" Color="#ffffff" />

		
		<MauiSplashScreen Include="Resources\Splash\splash.svg" Color="#ffffff" BaseSize="128,128" />

		
		<MauiImage Include="Resources\Images\*" />
		<MauiImage Update="Resources\Images\dotnet_bot.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\aboutus.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\appointment.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\banner1.jpg" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\banner2.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\banner3.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\bg_1.jpg" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\bg_2.jpg" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\booked.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\cancelled.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\choosrservice.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\confirmpayment.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\ecare.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\empty.jpg" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\history.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\kihnewlogo.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\kihsmlogo.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\login.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\loginbtn.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\logout.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\logowithtext.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\mail.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\nointernet.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\password.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\prescription.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\profile.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\report.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\somethingwrong.jpg" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\time.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\user.png" Resize="True" BaseSize="300,185" />
		<MauiImage Update="Resources\Images\userprof.png" Resize="True" BaseSize="300,185" />

		
		<MauiFont Include="Resources\Fonts\*" />
		
		
		<MauiAsset Include="Resources\Raw\**" LogicalName="%(RecursiveDir)%(Filename)%(Extension)" />
	</ItemGroup>

	

	<ItemGroup>
		<PackageReference Include="Microsoft.Maui.Controls" Version="$(MauiVersion)" />
		<PackageReference Include="Microsoft.Maui.Controls.Compatibility" Version="$(MauiVersion)" />
		<PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="8.0.0" />
		<PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
		<PackageReference Include="Syncfusion.Licensing" Version="25.1.35" />
		<PackageReference Include="Syncfusion.Maui.Buttons" Version="25.1.35" />
		<PackageReference Include="Syncfusion.Maui.Carousel" Version="25.1.35" />
		<PackageReference Include="Syncfusion.Maui.Core" Version="25.1.35" />
		<PackageReference Include="Syncfusion.Maui.ImageEditor" Version="25.1.35" />
		<PackageReference Include="Syncfusion.Maui.ListView" Version="25.1.35" />
		<PackageReference Include="Syncfusion.Maui.PdfViewer" Version="25.1.35" />
	</ItemGroup>

</Project>

and MAUI.program 

namespace PatientApp
{
    public static class MauiProgram
    {
        public static MauiApp CreateMauiApp()
        {
            var builder = MauiApp.CreateBuilder();
            builder
                .UseMauiApp<App>()
                .ConfigureSyncfusionCore()
                .ConfigureFonts(fonts =>
                {
                    fonts.AddFont("OpenSans-Regular.ttf", "OpenSansRegular");
                    fonts.AddFont("OpenSans-Semibold.ttf", "OpenSansSemibold");
                });

#if DEBUG
    		builder.Logging.AddDebug();
#endif
            builder.Logging.AddDebug();
           
            return builder.Build();
        }
    }
}

**Comment by sadrirammal** [comment]
Hey There mate

I've been having the same issue and I found the solution. It is because of syncfusion. I was using an SFButton in my Xaml where I had 
<SF:Button>
<Image/>
</Sf:Button> 

this would work in debug mode but not in release mode. I already told Syncfusion support about it but they cant seem to find the issue. 

The way I found it was by commenting out all code one by one and then narrowing down the crashing code. This also has been happening with SF:Charts. 

Good luck mate

**Reply by UmairHashmi90** [reply]
I  resolve my issue basically issue is occur due styles.xml when i remove styles.xml from APP.xaml  my app is working fine

 ` <Application.Resources>
     <ResourceDictionary>
         <ResourceDictionary.MergedDictionaries>
             <ResourceDictionary Source="Resources/Styles/Colors.xaml" />
             
             <ResourceDictionary Source="Resources/Styles/CustomStyle.xaml" />
             <ResourceDictionary Source="Resources/Styles/FontIcon.xaml" />
         </ResourceDictionary.MergedDictionaries>
     </ResourceDictionary>
 </Application.Resources>`

**Comment by DDHSchmidt** [comment]
[AOT is known to screw things up in certain scenarios on iOS](https://github.com/dotnet/maui/issues/13019).
I don't know about Android but it might be worth checking out the [mentioned workaround](https://github.com/dotnet/maui/issues/13019#issuecomment-1528853032).
Otherwise: Try to get more output via adb when your app is crashing.
